.werft/delete-preview-environments/delete-preview-environment.ts

```diff
@@ -0,0 +1,50 @@
+import { Werft } from '../util/werft';
+import * as fs from 'fs';
+import * as Tracing from '../observability/tracing';
+import { SpanStatusCode } from '@opentelemetry/api';
+import { wipePreviewEnvironmentAndNamespace, helmInstallName } from '../util/kubectl';
+
+// Will be set once tracing has been initialized
+let werft: Werft
+
+const context = JSON.parse(fs.readFileSync('context.json').toString());
+
+Tracing.initialize()
+    .then(() => {
+        werft = new Werft("delete-preview-environment")
+    })
+    .then(() => deletePreviewEnvironment())
+    .then(() => werft.endAllSpans())
+    .catch((err) => {
+        werft.rootSpan.setStatus({
+            code: SpanStatusCode.ERROR,
+            message: err
+        })
+        werft.endAllSpans()
+    })
+
+async function deletePreviewEnvironment() {
+    werft.phase("preparing deletion")
+    const version = parseVersion();
+    const namespace = `staging-${version.split(".")[0]}`
+    werft.log("preparing deletion", `Proceeding to delete the ${namespace} namespace`)
+    werft.done("preparing deletion")
+
+    werft.phase("delete-preview")
+    try {
+        await wipePreviewEnvironmentAndNamespace(helmInstallName, namespace, { slice: 'delete-preview' })
+    } catch (err) {
+        werft.fail("delete-preview", err)
+    }
+    werft.done("delete-prewview")
+}
+
+
+export function parseVersion() {
+    let version = context.Name;
```

Comment:
Are you sure the name of the namespace can be derived by the Job name?

I don't see any delete-preview-* jobs on https://werft.gitpod-dev.com/ - did you run this? :) 

I was expecting you'd have to get the namespace based on Repository.ref but I might be wrong.

Reply:
I just copied this part from `build.ts`.

Over there the namespace is being derived from the job name 🤔
I believe because werft already does the name sanitization (get rid of unsupported characters)